doc/developers/contributing.rst

```diff
@@ -251,11 +251,21 @@ If any of the above seems like magic to you, then look up the `Git documentation
 web, or ask a friend or another contributor for help.
 
 If some conflicts arise between your branch and the ``master`` branch, you need
-to merge ``master``. The command will be::
+to merge ``master``. For that, you first need to fetch the ``upstream``'s
+``master``, and then merge it into your branch::
 
-  $ git merge master
+  $ git fetch upstream
+  $ git merge upstream/master
 
-with ``master`` being synchronized with the ``upstream``.
+It is recommended to work on temporary branches rather than directly on
+``master``. Since these temporary branches are branched off ``master``, it is
+helpful to keep your local ``master`` up to date with the upstream's
```

Comment:
Maybe rather than advising this, we should just recommend creating feature branches with:

```
git fetch upstream
get checkout -b myfeaturename upstream/master
```

Reply:
done!